src/kubernetes.cc

```diff
@@ -400,29 +389,30 @@ KubernetesReader::GetPodAndContainerMetadata(
   const json::Array* container_statuses =
       status->Get<json::Array>("containerStatuses");
 
-  // Move the container specs into a map from name to spec.
-  std::map<std::string, const json::Object*> container_spec_by_name;
-  for (const json::value& c_spec : *container_specs) {
-    const json::Object* container_spec = c_spec->As<json::Object>();
-    const std::string name = container_spec->Get<json::String>("name");
-    container_spec_by_name.emplace(name, container_spec);
+  // Move the container statuses into a map from name to status.
+  std::map<std::string, const json::Object*> container_status_by_name;
+  for (const json::value& c_status : *container_statuses) {
+    const json::Object* container_status = c_status->As<json::Object>();
+    const std::string name = container_status->Get<json::String>("name");
+    container_status_by_name.emplace(name, container_status);
   }
 
-  for (const json::value& c_status : *container_statuses) {
+  for (const json::value& c_spec : *container_specs) {
     if (config_.VerboseLogging()) {
-      LOG(INFO) << "Container: " << *c_status;
+      LOG(INFO) << "Container: " << *c_spec;
     }
-    const json::Object* container_status = c_status->As<json::Object>();
-    const std::string name = container_status->Get<json::String>("name");
-    auto spec_it = container_spec_by_name.find(name);
-    if (spec_it == container_spec_by_name.end()) {
-      LOG(ERROR) << "Internal error; spec not found for container " << name;
-      continue;
+    const json::Object* container_spec = c_spec->As<json::Object>();
+    const std::string name = container_spec->Get<json::String>("name");
+    auto status_it = container_status_by_name.find(name);
```

Comment:
What does "_it" mean here? "is there"?

Reply:
Short for "iterator". Fairly standard C++ convention.

Reply:
SG